catowseriOS/catowser/Toolbar/BrowserToolbarView.swift

```diff
@@ -295,10 +314,41 @@ private extension BrowserToolbarView {
         }
         rotate.startAnimation()
     }
+    
+    @available(iOS 17.0, *)
+    @MainActor
+    func startTabsObservation() {
+        withObservationTracking {
+            _ = UIServiceRegistry.shared().tabsSubject.selectedTabId
+        } onChange: {
+            Task { [weak self] in
```

Comment:
figure out if observation tracking can be defaulted to Main actor to avoid new task